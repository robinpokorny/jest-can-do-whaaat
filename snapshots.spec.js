const Immutable = require('immutable');
const { toMatchDiffSnapshot } = require('snapshot-diff');
expect.extend({ toMatchDiffSnapshot });

const plugin = {
  test(val) {
    return val && val.isPlay;
  },
  serialize(val, config, indent, depth, refs, printer) {
    const name = val.constructor.name;
    const newIndent = indent + config.indent;
    return (
      `Play <${val.title}>: ` +
      printer(val.content, config, newIndent, depth++, refs)
    );
  }
};

expect.addSnapshotSerializer(plugin);

describe('method', () => {
  test('works', () => {
    const full = {
      number: 1975,
      bool: true,
      string: 'Hello, Vanek',
      promise: Promise.resolve('Better job'),
      symbol: Symbol('brewery'),
      [Symbol.for('brewmaster')]: 'Next beer!',
      map: new Map([['position1', 'workman'], ['position2', 'stockkeeper']]),
      set: new Set(['think', 'write', 'snitch']),
      func: compromise => compromise * 2,
      null: null,
      undefined: undefined
    };

    expect(full).toMatchSnapshot('new name');
    expect(1936).toMatchSnapshot();
  });

  test('immutable', () => {
    expect(Immutable.Map({ a: 1, b: 2 })).toMatchSnapshot();
    expect(Immutable.List([2, 5, 11])).toMatchSnapshot();
  });

  test('mock function', () => {
    const fn = jest.fn();

    fn('Vanek');
    fn('Ferdinand', 'Vanek');

    expect(fn).toHaveBeenCalledWith('Vanek');
    expect(fn).toHaveBeenCalledWith('Ferdinand', 'Vanek');
    expect(fn).toHaveBeenCalledTimes(2);

    // vs

    expect(fn.mock.calls).toMatchSnapshot();
  });

  test('play', () => {
    const play = {
      isPlay: true,
      title: 'Audience',
      content: { scenes: 1 }
    };

    expect([play]).toMatchSnapshot();
  });

  test('diff', () => {
    const play = {
      title: 'Audience',
      characters: 2
    };

    expect(play).toMatchSnapshot();
    expect({ ...play, characters: 3 }).toMatchSnapshot();

    expect(play).toMatchDiffSnapshot({ ...play, characters: 3 });
  });
});
