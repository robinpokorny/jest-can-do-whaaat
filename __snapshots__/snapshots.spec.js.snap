// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`method diff 1`] = `
Object {
  "characters": 2,
  "title": "Audience",
}
`;

exports[`method diff 2`] = `
Object {
  "characters": 3,
  "title": "Audience",
}
`;

exports[`method diff 3`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"characters\\": 2,
+   \\"characters\\": 3,
    \\"title\\": \\"Audience\\",
  }"
`;

exports[`method immutable 1`] = `
Immutable.Map {
  "a": 1,
  "b": 2,
}
`;

exports[`method immutable 2`] = `
Immutable.List [
  2,
  5,
  11,
]
`;

exports[`method mock function 1`] = `
Array [
  Array [
    "Vanek",
  ],
  Array [
    "Ferdinand",
    "Vanek",
  ],
]
`;

exports[`method play 1`] = `
Array [
  Play <Audience>: Object {
      "scenes": 1,
    },
]
`;

exports[`method works 1`] = `1936`;

exports[`new name 1`] = `
Object {
  "bool": true,
  "func": [Function],
  "map": Map {
    "position1" => "workman",
    "position2" => "stockkeeper",
  },
  "null": null,
  "number": 1975,
  "promise": Promise {},
  "set": Set {
    "think",
    "write",
    "snitch",
  },
  "string": "Hello, Vanek",
  "symbol": Symbol(brewery),
  "undefined": undefined,
  Symbol(brewmaster): "Next beer!",
}
`;
